pkg/services/ngalert/state/state.go

```diff
@@ -134,3 +160,43 @@ func (a *State) setEndsAt(alertRule *ngModels.AlertRule, result eval.Result) {
 		a.EndsAt = result.EvaluatedAt.Add(time.Duration(alertRule.IntervalSeconds*2) * time.Second)
 	}
 }
+
+func createAlertAnnotation(new eval.State, alertRule *ngModels.AlertRule, result eval.Result) error {
+	dashUid, ok := alertRule.Annotations["__dashboardUid__"]
+	if !ok {
+		return nil
+	}
+
+	panelUid := alertRule.Annotations["__panelId__"]
+
+	panelId, err := strconv.ParseInt(panelUid, 10, 64)
```

Comment:
I wonder what would happen if the panel has been deleted. Most probably nothing except one redundant entry in the annotations table.
Note to myself: check and warn for linked rules when deleting a panel/dashboard.

Reply:
If you delete the entire dashboard, the annotation fails with an error saying that it can't find the dashboard (expected) if you remove the panel, it continues to create annotations because, currently, annotations are being created for every panel on a dashboard.